primitives/runtime-interface/src/impls.rs

```diff
@@ -195,12 +192,12 @@ impl<T: 'static + Decode> FromFFIValue for Vec<T> {
 	fn from_ffi_value(arg: u64) -> Vec<T> {
 		let (ptr, len) = unpack_ptr_and_len(arg);
 		let len = len as usize;
+		let data = unsafe { Vec::from_raw_parts(ptr as *mut u8, len, len) };
```

Comment:
Not related to this PR thus not to address here, but is there a guarantee that `len` > 0?

Reply:
I will add a check, but I think it should be okay without it as well: https://doc.rust-lang.org/1.4.0/src/alloc/raw_vec.rs.html#424

Reply:
Ah, no, I more thought about that with zero len often comes a zero pointer

https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=5d9e82f1097c43c090f73e1c37ceeb22

I think in order to avoid this pitfall and also instead of trying to ensure that `ptr` is actually has a proper alignment for `T` it is just easier to explicitly handle the zero length case.

Reply:
Yeah it is handled now, but WTF. Why does `reconstruct` returns a `None`?

Reply:
`Vec` can be thought as if it has a `data_ptr: NonNull` field. If you actually put null there it is UB.

That demo consistently exhibits such a behavior, because Rust uses layout optimization for enums 
based on illegal values. That's why `ptr::null()` maps to `None` in `Option<NonNull<T>>`.

Reply:
Interesting :)